src/app/view/perspectives/search/views/view-detail/view-detail.component.ts

```diff
@@ -59,6 +64,13 @@ export class ViewDetailComponent implements OnInit, OnChanges {
 
   public readonly viewType = ResourceType.View;
 
+  public sizeXL = SizeType.XL;
+  public sizeL = SizeType.L;
+  public sizeM = SizeType.M;
+  public sizeS = SizeType.S;
```

Comment:
Why these reduntant constants? You can define constant like this: `public readonly sizeType = SizeType` and then use it with any value in template.